front/lib/api/assistant/conversation.ts

```diff
@@ -624,16 +624,201 @@ export async function* retryAgentMessage(
   | AgentGenerationSuccessEvent
   | AgentMessageSuccessEvent
 > {
+  const messageRow = await Message.findOne({
+    where: {
+      conversationId: conversation.id,
+      id: message.id,
+    },
+    include: [
+      {
+        model: AgentMessage,
+        as: "agentMessage",
+        required: true,
+      },
+    ],
+  });
+
+  if (!messageRow) {
+    return {
```

Comment:
Same as previous PR return yield?

Reply:
return in a generator finishes the generator with the final value given (which is the behaviour I was looking for): https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/function*#return_statement_in_a_generator

Not sure what "return yield" does and I couldn't find compelling explanations online, this is why I'm hesitant to use it -- but no prob to switch to that if you think it's better